TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -2,169 +2,132 @@ const Parser = require( './Parser' );
 const BenchmarkMetricRouter = require( './BenchmarkMetricRouter' );
 const BenchmarkMetric = require( './BenchmarkMetric' );
 
-const benchmarkDelimiterRegex = /\*\*\*\*\*\*\*\*\*\* START OF NEW TESTCI BENCHMARK JOB \*\*\*\*\*\*\*\*\*\*[\r\n]/;
-const benchmarkNameRegex = /Benchmark Name: (.*) Benchmark Variant: .*[\r\n]/;
-const benchmarkVariantRegex = /Benchmark Name: .* Benchmark Variant: (.*)[\r\n]/;
-const productResourceRegex = /Product Resource: (.*)[\r\n]/;
-
 class BenchmarkParser extends Parser {
 
-    // True iff all benchmarks and their corresponding iterations in the given run can be parsed
-    static canParse( buildName, output ) {
-        return benchmarkDelimiterRegex.test(output);
+    static getBenchmarkInfo( testName ) {
+        let benchmarkName, benchmarkVariant;
+
+        /* For PerfNext, benchmark name is allowed to have multiple underscores but variant shouldn't have underscores
+         * in order for TRSS to parse the job. For TKG, we'll be good since we don't specify variant name anyway.
+         */
+        if (testName) {
+            let index = testName.lastIndexOf("_")
+            if(index >= 0){
+                benchmarkName = testName.substring(0, index);
+                benchmarkVariant = testName.substring(index+1, testName.length);
+            }
+        }
+        return {benchmarkName, benchmarkVariant};
     }
 
-    // Returns an iterator object which accesses each benchmark iteration from the given output
-    static getBenchmarkIterator(output) {
-        let index = 1;
-        let splitOutput = output.split(benchmarkDelimiterRegex);
-        let splitOutputLength = splitOutput.length;
-    
-        // No delimiter inside the output
-        if (!(Array.isArray(splitOutput)) || splitOutputLength === 1) {
-            return null;
-        }
-        return {
-            next: function() {
-                return index < splitOutputLength ?
-                    {value: splitOutput[index++], done: false} :
-                    {done: true};
+    // True iff all benchmarks and their corresponding iterations in the given run can be parsed
+    static canParse( testName ) {
+        const {benchmarkName, benchmarkVariant} = this.getBenchmarkInfo(testName);
+        const variantValue = parseInt(benchmarkVariant);
+        if (benchmarkName && benchmarkVariant) {
```

Comment:
- Please check `BenchmarkMetricRouter[benchmarkName]` first, then check for the rest.
- Please call ` parseInt(benchmarkVariant)`  after verifying `BenchmarkMetricRouter[benchmarkName]`